source/adapters/level_zero/memory.cpp

```diff
@@ -2181,9 +2187,14 @@ ur_result_t _ur_buffer::getZeHandle(char *&ZeHandle, access_mode_t AccessMode,
         }
       }
       if (waitlist.Length > 0) {
-        ZE2UR_CALL(zeCommandListAppendWaitOnEvents,
-                   (UrContext->ZeCommandListInit, waitlist.Length,
-                    waitlist.ZeEventList));
+        ze_command_list_handle_t ZeCommandListForInit =
+            UrContext->ZeCommandListInit;
+        if (InOrderListRequired) {
+          ZeCommandListForInit = UrContext->ZeCommandListInitInOrder;
```

Comment:
can we just create the correct type of cmdlist in the first place in `UrContext->ZeCommandListInit`? Will there ever be a situation where two different queues in a single context require different types of events (thus command lists)?

Reply:
the user can request an out of order queue so the envs would not be taken so we would have an out of order queue, then an in order immediate command list being used for init.

There is no "correct" in this case. They would be different in this case anyways.

If we want the init list to always be determined by the defaults of the two envs, then  we can change this such that is the envs are set, then the init command list is also in order.

I was worried about the impact of performance for these calls, but if that is not a concern, then this can be made that way.

Reply:
@pbalcer , please re-review, I decided to just simplify to the single list when both envs are set. that would then be a catch all. Less configurable to swap inbetween, but this is cleaner.

Reply:
lgtm, thanks for making the changes!